### PR TITLE
Create Proposal UI Redesign (Safe/Usul)

### PR DESCRIPTION
**Pull request by mobiusTripper-crypto**
## Description

Updated Create Proposal page to match the next UX design documents.

## Notes

The pages for this branch do not match the designs until it is merged with `update/fractal-design-0.1`, but the components function correctly.
Additionally, the panel on the right side `Proposal Details` is not completed and needs to have live added added in a future PR. 
See included inline review comments

## Testing

- Navigate to an existing DAO [0xB87AD11E40F66137BD4cbB5f9d9438451000251B]. View all proposals.
- In the upper left of the screen choose `Create Proposal` to create a new proposal.
- Verify Accordion functions operate correctly for expand / collapse (click the Transaction number header)
- Verify Add and Remove transactions work correctly.

### CODE REVIEW

File: src/components/ProposalCreate/InputComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import { Input, HStack, Text, Textarea, Grid, GridItem } from '@chakra-ui/react';
```
Comment (mobiusTripper-crypto) [inline src/components/ProposalCreate/InputComponent.tsx:1]: This violates the "1 component 1 file" guideline. InputComponent and TextareaComponet can be separated, but I wanted some initial feedback on this design approach.

File: src/components/ProposalCreate/Transactions.tsx [hunk 2]
```diff
@@ -20,20 +32,84 @@ function Transactions({
...
+                {index !== 0 ? (
+                  <IconButton
+                    icon={<DeleteIcon />}
+                    aria-label={t('removetransactionlabel')}
+                    variant="unstyled"
+                    onClick={() => removeTransaction(index)}
+                    minWidth="auto"
+                  />
+                ) : (
+                  <Box h="36px" />
```
Comment (mobiusTripper-crypto) [inline src/components/ProposalCreate/Transactions.tsx:97]: I needed a placeholder here when the icon is not displayed. Is this an acceptable solution?

File: src/i18n/locales/en/proposal.json [hunk 3]
```diff
@@ -66,5 +66,9 @@
     "filterProposalsNSelected": "Filter {{count}}",
     "filter": "Filter",
     "parent": "Parent",
-    "child": "Child"
+    "child": "Child",
+    "cancel": "Cancel",
```
Comment (mobiusTripper-crypto) [inline src/i18n/locales/en/proposal.json:70]: I was unclear if these single word translations should go in 'common'

File: src/pages/ProposalCreate/index.tsx [hunk 4]
```diff
@@ -118,85 +123,115 @@ function ProposalCreate() {
...
+                borderColor="#0085FF"
+                status="error"
+                bg="#152023"
+                w="100%"
+                borderRadius={8}
+              >
+                <AlertIcon color="#0085FF">
+                  <Info
+                    boxSize="1.5rem"
+                    color="#0085FF"
```
Comment (herbig) [inline src/pages/ProposalCreate/index.tsx:193]: Do we not have this color in the design system? @Da-Colon ?

File: src/pages/ProposalCreate/index.tsx [hunk 5]
```diff
@@ -118,85 +123,115 @@ function ProposalCreate() {
...
+              >
+                {t('createProposal')}
+              </Button>
+            </VStack>
+          </ContentBox>
+        </VStack>
+      </GridItem>
+      <GridItem area="details">
+        <ContentBox bg="black.900-semi-transparent">
+          <Text textStyle="text-2xl-mono-bold">{t('proposalDetails')}</Text>
```
Comment (herbig) [inline src/pages/ProposalCreate/index.tsx:221]: doesn't look like `proposalDetails` is defined anywhere.  We have this as `proposalSummaryTitle` though

File: src/pages/ProposalCreate/index.tsx [hunk 6]
```diff
@@ -35,6 +47,7 @@ function ProposalCreate() {
    * adds new transaction form
    */
   const addTransaction = () => {
+    console.log('addnew', transactions.length);
```
Comment (herbig) [inline src/pages/ProposalCreate/index.tsx:50]: just noting we should remove stray console logs!

Also (not applicable to this instance), but we have a `logError(e);` function to log things to sentry/console on non prod builds, just FYI

File: src/i18n/locales/en/proposal.json (hunk 3 above)
Comment (herbig) [reply src/i18n/locales/en/proposal.json:68]: Hmm, yeah it seems like `cancel` definitely could be in common, that will likely be reused.  The others are probably fine to leave here

File: src/components/ProposalCreate/Transactions.tsx (hunk 2 above)
Comment (herbig) [reply src/components/ProposalCreate/Transactions.tsx:97]: yeah, makes sense to me, @Da-Colon might have ideas for alternatives

**Review by herbig — COMMENTED** [review]
Hey @mobiusTripper-crypto, this is looking super slick.

Some non code related feedback (all on _Firefox_, if that's relevant):

- margin between "Proposal" title and the black box "Transaction 1" looks thin
- Aside from Target Address, the background of the inputs are black, when they should be grey
- Inputting the create URL directely (e.g. https://deploy-preview-718--fractal-framework-interface-dev.netlify.app/#/daos/0xB87AD11E40F66137BD4cbB5f9d9438451000251B/proposals/new) then clicking "Cancel" gives a blank page
- is it possible to close the above transaction and/or scroll to the new one if "Add another transaction" is clicked?
- can we disable "Create Proposal" if clicking it won't start a transaction (meaning some of the fields aren't filled out)
- despite what the design shows, `Function Signature` and `Parameters` aren't required fields, since a function call could have no parameters

**Comment by mobiusTripper-crypto** [comment]
> * margin between "Proposal" title and the black box "Transaction 1" looks thin
fixed

> * Aside from Target Address, the background of the inputs are black, when they should be grey
the style of the textareas will be updated on the next revision of decent-ui

> * Inputting the create URL directely (e.g. https://deploy-preview-718--fractal-framework-interface-dev.netlify.app/#/daos/0xB87AD11E40F66137BD4cbB5f9d9438451000251B/proposals/new) then clicking "Cancel" gives a blank page
yeah, i totally phoned that in, fixed

> * is it possible to close the above transaction and/or scroll to the new one if "Add another transaction" is clicked?
yeah, good call here too, fixed

> * can we disable "Create Proposal" if clicking it won't start a transaction (meaning some of the fields aren't filled out)
I looked into this, and it might be a significate refactor from it's current implementations. maybe a new task would be a better idea and fix in this PR.

> * despite what the design shows, `Function Signature` and `Parameters` aren't required fields, since a function call could have no parameters
fixed

**Comment by herbig** [comment]
> > * can we disable "Create Proposal" if clicking it won't start a transaction (meaning some of the fields aren't filled out)
> >   _**I looked into this, and it might be a significate refactor from it's current implementations. maybe a new task would be a better idea and fix in this PR.**_

Yeah, makes sense. This is probably a larger discussion about form validations, as screens are currently all over the place on that..

**Review by herbig — COMMENTED** [review]
Hey @mobiusTripper-crypto a few minor things now with the margin change I think:

- There's no margin under "Create Proposal" title
- The `Proposal Details` cards isn't aligned with the Create side
- unrelated, but I also noticed the `Proposal Details` text is too large

Screenshot 2022-11-30 at 6 42 57 PM

File: src/components/ProposalCreate/Transactions.tsx (hunk 2 above)
Comment (Da-Colon) [reply src/components/ProposalCreate/Transactions.tsx:97]: Is there any reason why this wouldn't work here?

```tsx
                {index !== 0 && (
                  <IconButton
                    icon={<DeleteIcon />}
                    aria-label={t('removetransactionlabel')}
                    variant="unstyled"
                    onClick={() => removeTransaction(index)}
                    minWidth="auto"
                  />
                )}
```

File: src/pages/ProposalCreate/index.tsx (hunk 4 above)
Comment (Da-Colon) [reply src/pages/ProposalCreate/index.tsx:207]: Yessir, I'm the background is "blue.900" and the Icon and Border are "blue.500". I am also currently themeing the alert and will have it a new release soon where some of these values can be removed.

File: src/components/ProposalCreate/Transactions.tsx [hunk 7]
```diff
@@ -20,20 +32,84 @@ function Transactions({
...
-          transactionNumber={index}
-          updateTransaction={updateTransaction}
-          removeTransaction={removeTransaction}
-          transactionCount={transactions.length}
-          pending={pending}
-        />
+          borderTop="none"
+          borderBottom="none"
+        >
+          {({ isExpanded }) => (
```
Comment (Da-Colon) [inline src/components/ProposalCreate/Transactions.tsx:71]: These Accordion components are nifty. I've could have used these in the past

File: src/components/ProposalCreate/Transactions.tsx [hunk 8]
```diff
@@ -20,20 +32,84 @@ function Transactions({
...
 
+  const { t } = useTranslation(['proposal', 'common']);
+
+  function toggleExpanded(transactionNumber: number) {
+    const newTransactionData = Object.assign({}, transactions[transactionNumber]);
+    newTransactionData.isExpanded = !transactions[transactionNumber].isExpanded;
+    updateTransaction(newTransactionData, transactionNumber);
+  }
+
+  //TODO: placeholder for delete icon, remove when added to decent-ui
```
Comment (Da-Colon) [inline src/components/ProposalCreate/Transactions.tsx:43]: This has been add to the Design System and will be incoming in next release.

File: src/components/ProposalCreate/InputComponent.tsx (hunk 1 above)
Comment (Da-Colon) [reply src/components/ProposalCreate/InputComponent.tsx:1]: This looks good. And changes my mind a little on deconstructing `props` outside of the params lol.

**Review by Da-Colon — CHANGES_REQUESTED** [review]
Looking great. Updates for next release might be done in time to install here, I'll keep you posted. a few comments and suggestions

File: src/components/ProposalCreate/Transactions.tsx (hunk 2 above)
Comment (mobiusTripper-crypto) [reply src/components/ProposalCreate/Transactions.tsx:97]: Regarding this minWidth="auto" /> ) : ( <Box h="36px" />

The IconButton happens to be the tallest component in the Accordion when it's collapsed, so on the 1st transaction where there is no icon, it's height smaller than all the rest of the boxes. I could set a minHeight on the accordion panel equal to that icon, but somehow I need to keep that box from getting too small.

Comment (Da-Colon) [reply src/components/ProposalCreate/Transactions.tsx:97]: I see. yeah if it makes things complicated then the way it is fine for now

**Pull request merged**